pkg/virtctl/vnc/novnc/server.go

```diff
@@ -0,0 +1,146 @@
+package novnc
+
+import (
+	"archive/zip"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"golang.org/x/tools/godoc/vfs"
+	"golang.org/x/tools/godoc/vfs/zipfs"
+	"kubevirt.io/client-go/kubecli"
+)
+
+//go:embed novnc.zip
+var novnc []byte
```

Comment:
This is a really neat way of providing vnc.

The primary thing I'm hung up on is whether we should be packaging up and embedding javascript in our project. This opens up a whole new chain of java script dependencies which then puts us in line for handling CVEs for javascript dependencies for this novnc logic.

So, I like this functionality a lot. I'm just unsure about maintaining it primarily because of all the embedded javascript.

Reply:
> If it is behind a build flag (btw. right now only bazel build adds the zip, was not planning to commit the zip archive directly in the source), then vendors can decide on their own if they want to do CVE work on this and if they find the feature valuable.

yep, a build flag works for me.